sql/core/src/test/resources/sql-tests/results/interval.sql.out

```diff
@@ -1459,3 +1459,83 @@ struct<>
 -- !query output
 java.lang.ArithmeticException
 not in range
+
+
+-- !query
+SELECT INTERVAL '-106751991' DAY
+-- !query schema
+struct<INTERVAL '-106751991 days':interval>
+-- !query output
+-106751991 days
+
+
+-- !query
+SELECT INTERVAL '-106751991 04' DAY TO HOUR
+-- !query schema
+struct<INTERVAL '-106751991 04' DAY TO HOUR:interval day to hour>
+-- !query output
+-106751991 04:00:00.000000000
+
+
+-- !query
+SELECT INTERVAL '-106751991 04:00' DAY TO MINUTE
+-- !query schema
+struct<INTERVAL '-106751991 04:00' DAY TO MINUTE:interval day to minute>
+-- !query output
+-106751991 04:00:00.000000000
+
+
+-- !query
+SELECT INTERVAL '-106751991 04:00:54.775808' DAY TO SECOND
+-- !query schema
+struct<INTERVAL '-106751991 04:00:54.775808' DAY TO SECOND:interval day to second>
+-- !query output
+-106751991 04:00:54.775808000
+
+
+-- !query
+SELECT INTERVAL '-2562047788' HOUR
+-- !query schema
+struct<INTERVAL '-2562047788 hours':interval>
+-- !query output
+-2562047788 hours
+
+
+-- !query
+SELECT INTERVAL '-2562047788:00' HOUR TO MINUTE
+-- !query schema
+struct<INTERVAL '-2562047788:00' HOUR TO MINUTE:interval hour to minute>
+-- !query output
+-106751991 04:00:00.000000000
+
+
+-- !query
+SELECT INTERVAL '-2562047788:00:54.775808' HOUR TO SECOND
+-- !query schema
+struct<INTERVAL '-2562047788:00:54.775808' HOUR TO SECOND:interval hour to second>
+-- !query output
+-106751991 04:00:54.775808000
+
+
+-- !query
+SELECT INTERVAL '-153722867280' MINUTE
+-- !query schema
+struct<INTERVAL '-2562047788 hours':interval>
+-- !query output
+-2562047788 hours
+
+
+-- !query
+SELECT INTERVAL '-153722867280:54.775808' MINUTE TO SECOND
+-- !query schema
+struct<INTERVAL '-153722867280:54.775808' MINUTE TO SECOND:interval minute to second>
+-- !query output
+-106751991 04:00:54.775808000
+
+
+-- !query
+SELECT INTERVAL '-9223372036854.775808' SECOND
+-- !query schema
+struct<INTERVAL '-2562047788 hours -54.775808 seconds':interval>
+-- !query output
+-2562047788 hours -54.775808 seconds
```

Comment:
This format looks different from others, is it correct ? @MaxGekk

Reply:
> is it correct ?

Don't think so. It seems when start field == end field, the output style is "old" instead of ANSI one. @AngersZhuuuu  Could you open a ticket in JIRA for that.

Reply:
> > is it correct ?
> 
> Don't think so. It seems when start field == end field, the output style is "old" instead of ANSI one. @AngersZhuuuu Could you open a ticket in JIRA for that.

Yea, so seems all  `start field == end field` 's result is not correct?

Reply:
At least not all combinations are handled here https://github.com/apache/spark/blob/b0b9643cd76da48ed90e958e40717a664bc7494b/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L2487-L2504

Reply:
Does the code duplicate castStringToDTInterval and castStringToYMInterval. If so, it would be nice to avoid such duplications.

Reply:
For this part, let me do it after  https://github.com/apache/spark/pull/32949

Reply:
> > is it correct ?
> 
> Don't think so. It seems when start field == end field, the output style is "old" instead of ANSI one. @AngersZhuuuu Could you open a ticket in JIRA for that.

After check the code https://github.com/apache/spark/pull/32949 will solve this, right?